beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go

```diff
@@ -53,38 +51,30 @@ func (vs *Server) getLocalPayload(ctx context.Context, blk interfaces.ReadOnlyBe
 	slot := blk.Slot()
 	vIdx := blk.ProposerIndex()
 	headRoot := blk.ParentRoot()
-	proposerID, payloadId, ok := vs.ProposerSlotIndexCache.GetProposerPayloadIDs(slot, headRoot)
-	feeRecipient := params.BeaconConfig().DefaultFeeRecipient
-	recipient, err := vs.BeaconDB.FeeRecipientByValidatorID(ctx, vIdx)
-	switch err == nil {
-	case true:
-		feeRecipient = recipient
-	case errors.As(err, kv.ErrNotFoundFeeRecipient):
-		// If fee recipient is not found in DB and not set from beacon node CLI,
-		// use the burn address.
-		if feeRecipient.String() == params.BeaconConfig().EthBurnAddressHex {
-			logrus.WithFields(logrus.Fields{
-				"validatorIndex": vIdx,
-				"burnAddress":    params.BeaconConfig().EthBurnAddressHex,
-			}).Warn("Fee recipient is currently using the burn address, " +
-				"you will not be rewarded transaction fees on this setting. " +
-				"Please set a different eth address as the fee recipient. " +
-				"Please refer to our documentation for instructions")
-		}
-	default:
-		return nil, false, errors.Wrap(err, "could not get fee recipient in db")
+	logFields := logrus.Fields{
+		"validatorIndex": vIdx,
+		"slot":           slot,
+		"headRoot":       fmt.Sprintf("%#x", headRoot),
+	}
+	payloadId, ok := vs.PayloadIDCache.PayloadID(slot, headRoot)
+
+	val, tracked := vs.TrackedValidatorsCache.Validator(vIdx)
+	if !tracked {
+		logrus.WithFields(logFields).Warn("could not find tracked proposer index")
 	}
 
-	if ok && proposerID == vIdx && payloadId != [8]byte{} { // Payload ID is cache hit. Return the cached payload ID.
-		var pid [8]byte
+	var err error
+	if ok && payloadId != [8]byte{} {
+		// Payload ID is cache hit. Return the cached payload ID.
+		var pid primitives.PayloadID
 		copy(pid[:], payloadId[:])
 		payloadIDCacheHit.Inc()
 		var payload interfaces.ExecutionData
 		var overrideBuilder bool
 		payload, fullBlobsBundle, overrideBuilder, err = vs.ExecutionEngineCaller.GetPayload(ctx, pid, slot)
 		switch {
 		case err == nil:
-			warnIfFeeRecipientDiffers(payload, feeRecipient)
+			warnIfFeeRecipientDiffers(payload, val.FeeRecipient)
```

Comment:
If `tracked` is false, then `val.FeeRecipient` will panic right?

Reply:
no, the cache returns a Validator, most probably will be the burn address though but still better to continue I think .